File: CycleLength/solve/cpp/CycleLengthSolver.cpp
#include <fstream>
#include <topcoder>
#include "CycleLength.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
  try {
    std::ifstream ifs(argv[1]);
    int seed; tc::read(ifs, seed); tc::next(ifs);
    int a; tc::read(ifs, a); tc::next(ifs);
    int b; tc::read(ifs, b); tc::next(ifs);
    int m; tc::read(ifs, m);
    ifs.close();

    std::ofstream ofs(argv[2]);
    CycleLength solver;
    tc::show(ofs, solver.solve(seed, a, b, m));
    ofs.close();
  } catch (std::exception &e) {
    std::cerr << e.what() << std::endl;
  }
  return 0;
}

File: CycleLength/solve/java/CycleLengthSolver.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class CycleLengthSolver {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        int seed = (Integer) reader.next(Integer.class);
        reader.next();
        
        int a = (Integer) reader.next(Integer.class);
        reader.next();
        
        int b = (Integer) reader.next(Integer.class);
        reader.next();
        
        int m = (Integer) reader.next(Integer.class);
        reader.close();

        CycleLength solver = new CycleLength();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.solve(seed, a, b, m));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }
}
